deploy/podman/README.md

```diff
@@ -1,24 +1,72 @@
-# Deploy with podman
+# Deploy with Podman
 
 In this scenario the service and associated components are deployed onto your
 local host in containers using [Podman](https://podman.io/).
 
-## Requirements
+## General Requirements
 
-Make sure you have [podman](https://podman.io) version 3.3+ installed. If you
+Make sure you have [Podman](https://podman.io) version 3.3+ installed. If you
 must use an older version of podman, reference the [previous documentation and
 procedure](https://github.com/openshift/assisted-service/tree/v2.0.11#deploy-without-a-kubernetes-cluster)
 to avoid a [podman bug](https://github.com/containers/podman/issues/9609).
 
-Grab [pod.yml](./pod.yml) and [configmap.yml](configmap.yml) from this
-directory. No need to clone the whole repo.
+## Requirements for an HTTP based deployment
+
+### Files
+No need to clone the whole repo. Use:
+* [pod.yml](./pod.yml)
+* [configmap.yml](configmap.yml)
+
+### Environment Variable
 
 Change `IMAGE_SERVICE_BASE_URL` and `SERVICE_BASE_URL` in
 [configmap.yml](./configmap.yml) to match the hostname or IP address of your
 host. For example if your IP address is 192.168.122.2, then the
 `SERVICE_BASE_URL` would be set to <http://192.168.122.2:8090>. Port 8090 is
 the assisted-service API that agents will connect to.
 
+## Requirements for an HTTPS based deployment
+
+### SSL Certificates
+
+:warning: It would be best if you used valid and signed certificates.
+Below is an example of how to generate self-signed certificates, which are *not recommended* for use in production.
+
+#### Generate
+```shell
+openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -keyout tls.key -out tls.crt -subj "/CN=localhost" -addext 'subjectAltName = IP:127.0.0.1'
+```
+#### Embed to a ConfigMap
+Embed both key and certificate in [configmap_ssl_certs.yml](configmap_ssl_certs.yml)
+Those will get mounted to `assisted-installer-service`, `assisted-installer-image-service` and `assisted-installer-ui` for encryption.
+
+### Files
+No need to clone the whole repo. Use:
+* [pod_ssl.yml](./pod_ssl.yml)
+* [configmap_ssl.yml](configmap_ssl.yml)
+* [configmap_ssl_certs.yml](configmap_ssl_certs.yml)
+
+### Environment Variable
+
+
+
+
+Change `IMAGE_SERVICE_BASE_URL` and `SERVICE_BASE_URL` in
+[configmap_ssl.yml](./configmap_ssl.yml) to match the hostname or IP address of your
+host. For example if your IP address is 192.168.122.2, then the
+`SERVICE_BASE_URL` would be set to <https://192.168.122.2:8090>. Port 8090 is
+the assisted-service API that agents will connect to.
+
+Additionally, make sure to change the following:
+```yaml
+ASSISTED_SERVICE_URL: https://127.0.0.1:8090  
+HTTPS_CERT_FILE: "/etc/certs/tls.crt" # Need to match certs-configmap-volume mountPath
+HTTPS_KEY_FILE: "/etc/certs/tls.key"  # Need to match certs-configmap-volume mountPath
+HTTPS_CA_FILE: "/etc/certs/tls.crt"   # Need to match certs-configmap-volume mountPath
```

Comment:
Maybe note that this is only necessary if the certificate is not signed by a valid CA?

Reply:
You mean line#65 specifically? or 63-65?

Reply:
Specifically 65. `HTTPS_CERT_FILE` and `HTTPS_KEY_FILE` are what the servers use to serve https. `HTTPS_CA_FILE` is what the image service uses to trust assisted-service when it queries.

Sorry for the confusing names :worried:

Reply:
@carbonin ack.
How about:
```yaml
HTTPS_CA_FILE: "/etc/certs/tls.crt" # Need to match certs-configmap-volume mountPath. Needed if the certificate is not signed by a valid CA.
```

Already pushed to this PR. Let me know if that works.